### PR TITLE
Forbid to vote for worker objects with same state #1139

### PR DESCRIPTION
**Pull request by maslenitsa93**
#1139

### CODE REVIEW

**Comment by afalaleev** [comment]
Fix conflicts

**Comment by maslenitsa93** [comment]
Now we have fully same logic as in techspec and result approves. But counting only disapproves. It is good and understandable.

File: libraries/chain/worker_evaluators.cpp [hunk 1]
```diff
@@ -180,14 +185,17 @@ namespace golos { namespace chain {
         auto wtao_itr = wtao_idx.find(std::make_tuple(wto.post, o.approver));
 
         if (o.state == worker_techspec_approve_state::abstain) {
-            if (wtao_itr != wtao_idx.end()) {
-                _db.remove(*wtao_itr);
+            if (wtao_itr == wtao_idx.end()) {
+                WORKER_CHECK_NO_VOTE_REPEAT(worker_techspec_approve_state::abstain, worker_techspec_approve_state::abstain);
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: `WORKER_CHECK_NO_VOTE_REPEAT(wtao_itr,  wtao_idx.end());` - looks more simple.

File: libraries/chain/worker_evaluators.cpp [hunk 2]
```diff
@@ -320,14 +328,17 @@ namespace golos { namespace chain {
         auto wrao_itr = wrao_idx.find(std::make_tuple(worker_result_post.id, o.approver));
 
         if (o.state == worker_techspec_approve_state::abstain) {
-            if (wrao_itr != wrao_idx.end()) {
-                _db.remove(*wrao_itr);
+            if (wrao_itr == wrao_idx.end()) {
+                WORKER_CHECK_NO_VOTE_REPEAT(worker_techspec_approve_state::abstain, worker_techspec_approve_state::abstain);
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: Simplify code.

File: libraries/chain/worker_evaluators.cpp [hunk 3]
```diff
@@ -456,23 +468,28 @@ namespace golos { namespace chain {
         const auto& wpao_idx = _db.get_index<worker_payment_approve_index, by_result_approver>();
         auto wpao_itr = wpao_idx.find(std::make_tuple(worker_result_post.id, o.approver));
 
-        if (o.state != worker_techspec_approve_state::disapprove) {
-            if (wpao_itr != wpao_idx.end()) {
-                _db.remove(*wpao_itr);
+        if (o.state == worker_techspec_approve_state::abstain) {
+            if (wpao_itr == wpao_idx.end()) {
+                WORKER_CHECK_NO_VOTE_REPEAT(worker_techspec_approve_state::abstain, worker_techspec_approve_state::abstain);
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: Simplify code.

**Pull request merged**